### PR TITLE
added basic live actors data

### PR DESCRIPTION
**Pull request by marius851000**
While I still have some willpower remaining today (and it's 8AM...)

### CODE REVIEW

File: headers/types/ground_mode/live_actor.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#ifndef HEADERS_LIVE_ACTOR_GROUND_MODE_H_
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:1]: Per convention, please use `HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_` (based on the full path)

File: headers/types/ground_mode/live_actor.h [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+#define HEADERS_LIVE_ACTOR_GROUND_MODE_H_
+
+struct LiveActor {
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:4]: The case convention for struct tags is snake_case. So `struct live_actor`.

File: headers/types/ground_mode/live_actor.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    uint16_t id;     /* The id of the Actor in the actor list */
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:5]: Can we use C++ style `//` comments?

File: headers/types/ground_mode/live_actor.h [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    uint16_t kind;   /* The specie of the Actor */
+    bool is_enabled; /* true when the actor is loaded, false otherwise (should be checked if unsure)
+                      */
+    undefined field3_0x5;
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:9]: Can we not number these? Just use the offset for unknown fields, i.e. `field_0x5`

File: headers/types/ground_mode/live_actor.h [hunk 5]
```diff
@@ -0,0 +1,94 @@
+#ifndef HEADERS_LIVE_ACTOR_GROUND_MODE_H_
+#define HEADERS_LIVE_ACTOR_GROUND_MODE_H_
+
+struct LiveActor {
+    uint16_t id;     /* The id of the Actor in the actor list */
+    uint16_t kind;   /* The specie of the Actor */
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:6]: specie -> species. Also, is this the monster ID, or something else? If it's the monster ID, let's use `struct monster_id_16` as the type.

File: headers/types/ground_mode/live_actor.h [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    bool is_enabled; /* true when the actor is loaded, false otherwise (should be checked if unsure)
+                      */
+    undefined field3_0x5;
+    uint16_t hanger;
+    uint8_t sector;
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:11]: Do we know what these fields are? Comments would be nice if possible.

File: headers/types/ground_mode/live_actor.h [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    undefined field50_0x14b;
+    uint32_t field51_0x14c;
+    int16_t field52_0x150;
+    bool field53_0x152;
+    undefined field54_0x153;
+    int16_t field55_0x154;
+    bool field56_0x156;
+    undefined field57_0x157;
+    int16_t field58_0x158;
+    int8_t direction;
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:65]: Is `struct direction_id_8` (see `enum direction_id`) appropriate here?

File: headers/types/ground_mode/live_actor.h [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    struct vec2_int coord_max; /* the bottom-right coordinate of the collision box of the actor */
+    undefined4 field63_0x16c;
+    undefined4 field64_0x170;
+    int8_t field65_0x174;
+    int8_t field66_0x175;
+    int16_t field67_0x176;
+    int16_t field68_0x178;
+    undefined field69_0x17a;
+    undefined field70_0x17b;
+    int32_t movement_related; /* not sure. seem to be 0 when not moving and 3 when moving most of
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:77]: For comments that don't fit on one line, I usually like to put them on the line above.

File: headers/types/ground_mode/live_actor.h [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    undefined field78_0x189;
+    undefined field79_0x18a;
+    undefined field80_0x18b;
+    undefined animation[196]; /* a struct containing data relating to animation */
+};
+ASSERT_SIZE(struct LiveActor, 592);
+
+typedef struct LiveActor LiveActorList[24];
+
+#endif
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:94]: Newline at the end for consistency with the other files.

File: headers/types/ground_mode/live_actor.h [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    undefined field76_0x187;
+    undefined field77_0x188;
+    undefined field78_0x189;
+    undefined field79_0x18a;
+    undefined field80_0x18b;
+    undefined animation[196]; /* a struct containing data relating to animation */
+};
+ASSERT_SIZE(struct LiveActor, 592);
+
+typedef struct LiveActor LiveActorList[24];
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:92]: Please don't use typedef structs; they introduce unnecessary obfuscation. To define a list, do:

```c
struct live_actor_list {
    struct live_actor actors[24];
};
ASSERT_SIZE(struct live_actor_list, 14208);
```

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/ground_mode/live_actor.h (hunk 6 above)
Comment (marius851000) [reply headers/types/ground_mode/live_actor.h:11]: I think that sector is related to whether they are loaded of not in a script, but I never used something different than the default. They can easily be edited in SkyTemple, and named like that. I'll ask.

File: headers/types/ground_mode/live_actor.h (hunk 7 above)
Comment (marius851000) [reply headers/types/ground_mode/live_actor.h:65]: It is.

File: headers/types/ground_mode/live_actor.h (hunk 5 above)
Comment (UsernameFodder) [reply headers/types/ground_mode/live_actor.h]: > Also, is this the monster ID, or something else?

So is the answer to this question no? `struct monster_id_16` isn't appropriate here?

File: headers/types/ground_mode/live_actor.h (hunk 6 above)
Comment (UsernameFodder) [reply headers/types/ground_mode/live_actor.h:11]: It's okay, if you don't know we don't have to bother right now. We can always come back and add comments if someone needs them.

**Comment by theCapypara** [comment]
Hanger and sector are names that come from debug outputs of the game. 

A sector is a layer in script scenes. From my observations in RAM they are often not actually filled with the actual value but just  -1 or 0 instead. I think they are just filled for for entities if a script command exists for their hanger that could remove sectors (meaning removing or hiding all entities on the layer.

Hanger are slots for scenes+scripts on the script engine.
1 is for enter scenes 
2 is for acting scenes
3 is for sub scenes
4 is for talking sub scenes of enter scripts
5 is for special purposes iirc. eg. partner.ssb
6 is for talking sub scenes of sub scenes.

https://github.com/SkyTemple/skytemple-ssb-debugger/blob/4860a98317cd25d4af226a058e6158ffaa3e21f5/skytemple_ssb_debugger/controller/ground_state.py#L193

For details I'd recommend looking at the Ground State tab in the SkyTemple Debugger while the game has some scenes loaded.

**Comment by marius851000** [comment]
Oops, totally forgot about this one. Will go back to that later today.

File: headers/types/ground_mode/live_actor.h [hunk 11]
```diff
@@ -0,0 +1,98 @@
+#ifndef HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_
+#define HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_
+
+// represent an actor present in the scene in the overworld (both during cutscenes and free-roams)
+struct live_actor {
+    struct monster_id_16 id; // The id of the Actor in the actor list
+    uint16_t kind;           // The species of the Actor
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:7]: Is this right? I don't actually know what these two fields represent other than what your comments say, so please correct me if I'm wrong.

`struct monster_id_16` corresponds to the Pokémon species, e.g., Bulbasaur, Ivysaur, etc. From the descriptions it seems like this should be:

```c
uint16_t id;               // The id of the Actor in the actor list
struct monster_id_16 kind; // The species of the actor
```

**Comment by marius851000** [comment]
*this is just rebasing I just did*

File: headers/types/ground_mode/live_actor.h [hunk 12]
```diff
@@ -0,0 +1,98 @@
+#ifndef HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_
+#define HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_
+
+// represent an actor present in the scene in the overworld (both during cutscenes and free-roams)
+struct live_actor {
+    struct monster_id_16 specie_id; // The id of the Actor in the actor list. Internally named type.
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:6]: specie_id -> species_id

File: headers/types/ground_mode/live_actor.h [hunk 13]
```diff
@@ -0,0 +1,98 @@
+#ifndef HEADERS_TYPES_GROUND_MODE_LIVE_ACTOR_H_
```
Comment (UsernameFodder) [inline headers/types/ground_mode/live_actor.h:1]: This doesn't feel substantial enough to split out into a separate file; there's only two structs here. Can we just put these definitions directly in `ground_mode.h`? That file isn't super long anyway.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/common/common.h [hunk 14]
```diff
@@ -694,6 +694,20 @@ struct adventure_log {
 };
 ASSERT_SIZE(struct adventure_log, 636);
 
+// a 2d uint (32bit) vector
+struct vec2_uint {
```
[the remark this answers is not in the document]
Comment (marius851000) [reply headers/types/common/common.h:698]: Hmmm... I see a vec2 used in EuclideanNorm and ClampComponentAbs, but no definition for them... I hope adding a difinition will not break thing.

[the remark this answers is not in the document]
Comment (UsernameFodder) [reply headers/types/common/common.h]: Nah those are just parameter names, so we should be fine.

**Review by UsernameFodder — APPROVED** [review]
Looks good, thanks!

**Pull request merged**